### PR TITLE
Remove multi-partition support from StreamPlatform test extension

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description
Remove the multi partition support in the StreamPlatform, and related extension. This is not necessary for testing the StreamPlatform code

## Related issues

related to https://github.com/camunda/zeebe/issues/9727

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java [hunk 1]
```diff
@@ -250,68 +239,41 @@ public void onRecovered(final ReadonlyStreamProcessorContext context) {
     }
     openFuture.join(15, TimeUnit.SECONDS);
 
-    final ProcessorContext processorContext =
+    processorContext =
         ProcessorContext.createStreamContext(streamProcessor, zeebeDb, storage, snapshot);
-    streamContextMap.put(logName, processorContext);
-    closeables.add(() -> streamContextMap.remove(logName).close());
+    closeables.add(() -> processorContext.close());
```
Comment (npepinpe) [inline engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java:244]:
```suggestion
    closeables.add(processorContext);
```

**Review by npepinpe — APPROVED** [review]
:+1: 

- :wrench: If we're doing clean up, there's some dead code in this class, and I think the context classes can probably be records (e.g. `LogContext`). Might be a good opportunity to just do that now.

e.g. `LogContext` can become just

```java
private record LogContext(SynchronousLogStream logStream) implements AutoCloseable {

    @Override
    public void close() {
      logStream.close();
    }
  }
```

`ProcessorContext` could be a record eventually as well - not sure how important the idempotency of the close method is :shrug:

**Comment by Zelldon** [comment]
Thanks @npepinpe :+1: I did some further cleanup. :)

bors r+

File: engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java [hunk 2]
```diff
@@ -404,7 +315,6 @@ public void close() throws Exception {
       if (runtimePath.toFile().exists()) {
         FileUtil.deleteFolder(runtimePath);
       }
```
Comment (npepinpe) [inline engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java:317]:
```suggestion
      FileUtil.deleteFolderIfExists(runtimePath);
```

Comment (npepinpe) [reply engine/src/test/java/io/camunda/zeebe/engine/util/StreamPlatform.java:317]: Though I guess it's too late :smile: We can do that later :shrug:

**Pull request merged**